sky/provision/docker_utils.py

```diff
@@ -197,6 +197,11 @@ def initialize(self) -> str:
 
         self._check_docker_installed()
 
+        # SkyPilot: Add the current user to the docker group
+        # if not already added.
+        self._run('id -nG $USER | grep -qw docker || '
+                  'sudo usermod -aG docker $USER')
```

Comment:
Can we keep this command running within other commands, e.g., _check_docker_installed, to reduce the ssh connection?

Reply:
Also, just curious, how does it compare with using `sudo docker` in all places `docker` is called. Asking because had the same issue before running docker on TPU instances.

Reply:
Lets merge this version first to unblock our user ;) filed an issue #4121 to keep track of this.